### DISCUSSION TITLE
Allow continuing a failed play by appending "no fail" mod

### DISCUSSION THREAD

**Discussion by Craftplacer** [ppy/osu#14590] [Ideas]
I think there should be an option in the fail screen to "Continue with No Fail" (with disclaimer that it will reduce the previously earned score).

Just a thought, I don't know whether this has been discussed before.

**Comment by frenzibyte** [comment]
While it may be something cool to have in some cases, I'm not sure it'll look fine to have as part of the "retry, exit" buttons during failure.

Assuming the intention here is for faster way of practicing, maybe something like a blue "practice" button that switches to a ["practice" mode](https://github.com/ppy/osu/discussions/13013) during fail would be more sensible and handy to have? (once practice mode becomes a thing of course)

**Comment by peppy** [comment]
If this was added, it wouldn't be "continue with nofail", but "continue without submitting".

**Reply by Craftplacer** [reply]
Any reason why the score shouldn't be submitted?

I mentioned NF because I thought the player should still have a chance to get a score (although, being nerfed with the NF score multiplier).

**Reply by peppy** [reply]
This has come up hundreds of times. Why would you ever choose to turn on NF if this was the case? It's just taking any kind of fun out of the choice.

**Reply by Striikerr** [reply]
There have been many times where I've played a song hoping to pass then failed to a single stream or something (I obviously don't know if I would have failed again, but on most maps, it'd usually be once a diff spike) But even though something like this would benefit me, I still wouldn't want this regardless, IMO it takes away the flow of setting a score. People already have a slight issue with being able to set ranked scores while pausing, I couldn't imagine how the community would react to something like this.

**Reply by abraker95** [reply]
> This has come up hundreds of times. Why would you ever choose to turn on NF if this was the case? It's just taking any kind of fun out of the choice.

Maybe NF should be treated as an option instead of a mod. I don't see what is meant by "taking any kind of fun out of the choice". Players choose NF because there is one or more sections in the map where their skills are insufficient and either want to submit a score despite that or want to practice the map despite not being able to pass it. It's being treated as a gameplay setting, not as alternative gameplay mechanic.

**Reply by smoogipoo** [reply]
Yes but why would anyone play with "auto-nofail" turned off? It's the best of both worlds at top levels, no?
a) If you fail, welp that sucks, but you still get pp.
b) If you don't fail then great! You get 100% pp.

The point is that having this setting is equivalent to removing fails altogether. Look at tournaments for example, now that NF doesn't have a negative multiplier, literally everyone in tournaments picks NF - it's on by default for host-picks and it's automatically turned on by players in FM pools. So I have reason to believe that my assertion is accurate if you have no punishment for turning NF on.

And before you say there is a punishment - there isn't if you have full combo (https://github.com/ppy/osu-performance/pull/127), and it's a very small punishment otherwise (2% of pp for one miss...).

**Reply by Craftplacer** [reply]
I see with the points given that it indeed makes no sense to score the play or use NF at all, that being said, I'm open to the alternatives proposed.

**Reply by abraker95** [reply]
> Yes but why would anyone play with "auto-nofail" turned off?

Players would still like the option for the game to fail them in certain cases if they are doing poorly. In fact the concept behind the [selective perfect request](https://osu.ppy.sh/community/forums/topics/141061) is to have better control over when the game stops you. "auto-nofail" will be turned off by players who don't mind the score not submitting and want to focus on improving instead.

> Look at tournaments for example, now that NF doesn't have a negative multiplier, literally everyone in tournaments picks NF

No kidding, it's strategically beneficial to pick NF in tournaments so that whatever score you have still counts in case HP falls to 0. It's insurance.

**Reply by bdach** [reply]
> No kidding, it's strategically beneficial to pick NF in tournaments so that whatever score you have still counts in case HP falls to 0. It's insurance.

And this is somehow good? I beg to differ. Might as well remove the health mechanic from the game altogether at that point.

**Reply by abraker95** [reply]
> > No kidding, it's strategically beneficial to pick NF in tournaments so that whatever score you have still counts in case HP falls to 0. It's insurance.
> 
> And this is somehow good? I beg to differ. Might as well remove the health mechanic from the game altogether at that point.

Why do you think this is a bad thing? Also the health mechanic would still serve a purpose for players who would play with "auto-nofail" turned off, and I already stated why players would opt to turn it off.
